src/main/java/org/jooq/lambda/concurrent/Async.java

```diff
@@ -0,0 +1,20 @@
+package org.jooq.lambda.concurrent;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+import java.util.function.Supplier;
+
+public final class Async {
+    private Async() {
+
+    }
+
+    public static <U> CompletionStage<U> supplyAsync(Supplier<U> supplier, Executor executor) {
+        return SameExecutorCompletionStage.of(CompletableFuture.supplyAsync(supplier, executor), executor);
+    }
+
+    public static CompletionStage<Void> runAsync(Runnable runnable, Executor executor) {
+        return SameExecutorCompletionStage.of(CompletableFuture.runAsync(runnable, executor), executor);
+    }
```

Comment:
It depends of course.

- One one side, you're right. It's sometimes good to force people to think.
- On the other side, that's what the folks did with the good old XML APIs. You remember how hard it is to run a simple XPath query? Exactly. In many cases, I *don't* want to think too hard about all the builders, factories and what not.

I personally favour convenience here, along with the appropriate Javadoc, of course.